samples/Hystrix.Dotnet.Samples.Owin/Program.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Hystrix.Dotnet.Samples.Owin.Controllers;
+using Hystrix.Dotnet.WebConfiguration;
+using Microsoft.Owin.Hosting;
+using Owin;
+using SimpleInjector;
+using SimpleInjector.Lifestyles;
+using System;
+using System.Configuration;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace Hystrix.Dotnet.Samples.Owin
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var url = ConfigurationManager.AppSettings["url"];
+
+            if(string.IsNullOrWhiteSpace(url))
+            {
+                Console.WriteLine("You must provide \"url\" setting in config");
+                return;
+            }
+
+            using (WebApp.Start(url, Startup))
+            {
+                Console.WriteLine("Please call {0}/api/hello", url);
+                Console.WriteLine("Stream is here: {0}/hystrix.stream", url);
+                Console.Read();
+            }
+        }
+
+        static void Startup(IAppBuilder app)
+        {
+            var config = new HttpConfiguration();
+
+            config.Routes.MapHttpRoute(
+                name: "DefaultApi",
+                routeTemplate: "api/{controller}/{id}",
+                defaults: new { id = RouteParameter.Optional }
+            );
+
+            SetupDI(config);
+
+            app.UseHystrixStream();
+
+            app.UseWebApi(config);
+        }
+
+        static void SetupDI(HttpConfiguration conf)
+        {
+            var container = new Container();
+            container.Options.DefaultScopedLifestyle = new AsyncScopedLifestyle();
+
+            var helper = new AspNetHystrixCommandFactoryHelper();
+
+            var factory = helper.CreateFactory();
+
+            container.Register<GetBreakerCommand>(() =>
+                (cancel => GetBreakerCommand(factory.GetHystrixCommand("TestGroup", "TestCommand"), cancel)), Lifestyle.Scoped);
+
+            container.Verify();
+
+            conf.DependencyResolver =
+                new SimpleInjector.Integration.WebApi.SimpleInjectorWebApiDependencyResolver(container);
+        }
+
+        static readonly Random rnd = new Random();
+
+        static Task<string> GetBreakerCommand(IHystrixCommand hystrixCommand, CancellationToken cancellationToken)

```

Comment:
To me it doesn't feel like usual practice to set up an operation globally like this, I think it's usually more practical to set it up where we're actually doing it, in this case in the `HelloController`, could you move it there? Then the sample will be a bit simpler to understand too without the injection of a delegate.